pkg/codegen/codegen.go

```diff
@@ -462,6 +468,127 @@ func GenerateTypesForSchemas(t *template.Template, schemas map[string]*openapi3.
+	excludeSchemasMap := make(map[string]bool)
+	for _, schema := range excludeSchemas {
+		excludeSchemasMap[schema] = true
+	}
+
+	types := make([]TypeDefinition, 0)
+
+	// We're going to define Go types for every object under paths
+	for _, pathName := range SortedPathsKeys(paths) {
+		for _, operationName := range SortedOperationsKeys(paths[pathName].Operations()) {
+			operation := paths[pathName].Operations()[operationName]
+			// starting with request bodies
+			if operation.RequestBody != nil {
+				sortedContentKeys := SortedContentKeys(operation.RequestBody.Value.Content)
+				for _, mediaTypeStr := range sortedContentKeys {
+					mediaType := operation.RequestBody.Value.Content.Get(mediaTypeStr)
+
+					typeName := ""
+					if operation.OperationID != "" {
+						typeName += ToCamelCase(operation.OperationID)
+					} else {
+						typeName += ToCamelCase(strings.ToLower(operationName)) + ToCamelCase(pathName)
+					}
+					typeName += "RequestBody"
+
+					// when there are multiple media types provided for the Response, add a suffix for the media type
+					if len(sortedContentKeys) > 1 {
+						typeName += mediaTypeToGoName(mediaTypeStr)
+					}
+
+					goSchema, err := GenerateGoSchema(mediaType.Schema, []string{typeName})
+					if err != nil {
+						return nil, fmt.Errorf("error converting Schema for %s %s to Go type:: %w", operationName, pathName, err)
+					}
+
+					goTypeName, err := renameSchema(typeName, mediaType.Schema)
+					if err != nil {
+						return nil, fmt.Errorf("error generating a Go type name for %s %s with provided typeName %s: %w", operationName, pathName, typeName, err)
+					}
+
+					typeDef := TypeDefinition{
+						JsonName: fmt.Sprintf("the request body for %s %s (%s)", operationName, pathName, mediaTypeStr),
+						TypeName: goTypeName,
+						Schema:   goSchema,
+					}
+
+					if operation.RequestBody.Ref != "" {
+						// Generate a reference type for referenced types
+						refType, err := RefPathToGoType(operation.RequestBody.Ref)
+						if err != nil {
+							return nil, fmt.Errorf("error generating Go type for %s %s: %w", operationName, pathName, err)
+						}
+						typeDef.Schema = Schema{
+							RefType: refType,
+						}
+					}
+
+					types = append(types, typeDef)
+					types = append(types, goSchema.GetAdditionalTypeDefs()...)
+				}
+			}
+
+			// then with response bodies
+			for _, statusCodeOrDefault := range SortedResponsesKeys(operation.Responses) {
+				v := operation.Responses[statusCodeOrDefault]
+				sortedContentKeys := SortedContentKeys(v.Value.Content)
+				for _, mediaTypeStr := range sortedContentKeys {
+					mediaType := v.Value.Content.Get(mediaTypeStr)
+
+					typeName := ""
+					if operation.OperationID != "" {
+						typeName += ToCamelCase(operation.OperationID)
+					} else {
+						typeName += ToCamelCase(strings.ToLower(operationName)) + ToCamelCase(pathName)
+					}
+					typeName += fmt.Sprintf("ResponseBody%s", ToCamelCase(statusCodeOrDefault))
```

Comment:
Is this sufficient to avoid type conflict? What if a request has two response bodies with different content types for the same response code? What happens in that case?

Reply:
In that case, yep, it'll handle the conflict well enough

[i.e.](https://github.com/deepmap/oapi-codegen/pull/648/files#diff-6686d60bb4293c58e363e155ee4c6b689ce7fc4ab57109d44d691a3b0099bf8bR27-R31):

```go
// NewPetResponseBody200ApplicationYaml defines model for the response body for POST /pets (200, application/yaml).
type NewPetResponseBody200ApplicationYaml = map[string]interface{}

// NewPetResponseBody200TextPlain defines model for the response body for POST /pets (200, text/plain).
type NewPetResponseBody200TextPlain = map[string]interface{}
```